web/src/main/java/com/futurewei/alcor/web/rbac/aspect/RbacAspect.java

```diff
@@ -0,0 +1,238 @@
+ *         you may not use this file except in compliance with the License.
+ *         You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *         Unless required by applicable law or agreed to in writing, software
+ *         distributed under the License is distributed on an "AS IS" BASIS,
+ *         WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *         See the License for the specific language governing permissions and
+ *         limitations under the License.
+ * /
+ */
+
+package com.futurewei.alcor.web.rbac.aspect;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.futurewei.alcor.common.entity.TokenEntity;
+import com.futurewei.alcor.common.exception.PolicyNotAuthorizedException;
+import com.futurewei.alcor.common.exception.ResourceNotFoundException;
+import com.futurewei.alcor.common.rbac.RbacManger;
+import com.futurewei.alcor.common.utils.ControllerUtil;
+import com.futurewei.alcor.web.exception.NotAuthorizedException;
+import com.futurewei.alcor.web.exception.NotFoundException;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.*;
+
+import static com.futurewei.alcor.common.constants.CommonConstants.*;
+
+@Aspect
+@Component
+public class RbacAspect {
+    private static final Logger LOG = LoggerFactory.getLogger(RbacAspect.class);
+
+    private static final String POST_METHOD_NAME = "POST";
+    private static final String PUT_METHDO_NAME = "PUT";
+    private static final String DELETE_METHDO_NAME = "DELETE";
+    private static final String GET_METHDO_NAME = "GET";
+
+    @Autowired
+    private OwnerCheckerSupplier ownerCheckerSupplier;
+
+    @Autowired
+    private HttpServletRequest request;
+
+    @Autowired
+    private RbacManger rbacManger;
+
+    @Pointcut("@annotation(com.futurewei.alcor.web.rbac.aspect.Rbac)")
+    public void annotationPointCut(){}
+
+    @Around("annotationPointCut()")
+    public Object checkRbac(ProceedingJoinPoint pjp) throws Throwable {
+
+        String methodType = request.getMethod().toUpperCase();
+        String tokenInfo = request.getHeader(TOKEN_INFO_HEADER);
+        MethodSignature ms = (MethodSignature)pjp.getSignature();
```

Comment:
Could ms be possibly NULL or guaranteed to have value?

Reply:
> Could ms be possibly NULL or guaranteed to have value?
Ms could not be null.